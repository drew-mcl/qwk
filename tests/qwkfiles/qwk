MESSAGE = "Hello, World!"
COMMAND_VAR = $(pwd)
# - The following targets are used for testing the core functionality of qwk.
# - They expected to pass when qwk is working correctly.
# - For edge cases and tests that break core logic, see tests/qwkfiles/experimental & tests/qwkfiles/edge

# Chains
# --------------------------------------------------
all: single_command chained_commands variable_sub

# - Commands
# --------------------------------------------------
single_command:
	echo "Single command"

chained_commands:
	echo "Chained command 1" && echo "Chained command 2"

# - Variables
# --------------------------------------------------
variable_sub:
	echo $(MESSAGE)

command_sub:
	echo $(COMMAND_VAR)

# - Arguments
# --------------------------------------------------
single_argument:
	echo $1

multiple_arguments:
	echo $1 $2

arguments_inside_quotes:
	echo "First you said $1, then you said $2"

word_arguments:
	echo $*

word_arguments_inside_quotes:
	echo "$*"
