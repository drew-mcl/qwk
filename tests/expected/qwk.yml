Targets:
  all:
    single_command:
      output: "Single command"
    chained_commands:
      output: "Chained command 1\nChained command 2"
    variable_substitution:
      output: "Hello, World!"
